src/Controller/Api/QuapController.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace App\Controller\Api;
+
+use App\DTO\Model\WidgetControllerData\WidgetRequestData;
+use App\Entity\Group;
+use App\Exception\ApiException;
+use App\Service\QuapService;
+use DateTime;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\ParamConverter;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Exception\BadRequestException;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+
+class QuapController extends AbstractController
+{
+    public function getQuestionnaireData(
+        Request     $request,
+        QuapService $quapService,
```

Comment:
Please inject the quap service in a constructor
The injections inside the function should be endpoint specific. but this service is used from all endpoints.